sympy/geometry/line.py

```diff
@@ -2526,6 +2526,44 @@ def equation(self, x='x', y='y', z='z', k=None):
                 break
         return Tuple(*[i.subs(k, kk).as_numer_denom()[0] for i in eqs])
 
+    def bisection(self, line):
+        """Returns the line which is the bisection between the self line and another one.
```

Comment:
I would suggest the name "bisector" and say that it is the "line which bisects the angle formed at the intersection of the two lines and lying in the same plane"

But I wonder if, instead of raising an error for non-intersecting lines, a meaningful line could be returned which passes perpendicularly through the midpoint of the shortest segment connecting the two lines.
```python
>>> r1
Line3D(Point3D(0, 0, 0), Point3D(1, 1, 1))
>>> r4
Line3D(Point3D(1, 1, 0), Point3D(0, 2, 1))
>>> r1.intersection(r4)
[]
>>> t=Dummy()
>>> p1=r1.arbitrary_point(t)
>>> p4=r4.arbitrary_point(t)
>>> tmin=solve(p1.distance(p4).diff(t))[0] # find t which minimizes distance between arb pts
>>> m=Segment(p1,p4).midpoint.subs(t,tmin) # find mdpoint of shortest segment connecting lines
>>> d = Matrix(r1.direction_ratio).cross(Matrix(r4.direction_ratio)) # new direction
>>> Line(m,direction_ratio=d) # "bisector" line
Line3D(Point3D(1/2, 1, 1/2), Point3D(1/2, -1, 5/2))
```
So the definition of "bisector" for two lines is the line passing through either the intersection of the two lines or the line passing perpendicularly through the midpoint of the shortest segment between the two lines.

Reply:
Can you change the current code? Also the current code does not work with direction ratios greater than 90º and I don't know how to solve it.

Reply:
That is a good addition, but you are talking about a line perpendicular to a segment at certain point contained in the segment. There are infinite lines which satisfy that condition which define a plane.